teps/0094-improve-task-resources-ux.md

```diff
@@ -0,0 +1,119 @@
+
+## Motivation
+
+There are several issues affecting the user experience of specifying resource requirements for `Task`s:
+
+1. The way Tekton allows users to specify resource requirements is inconsistent with how `Task`s are actually scheduled,
+as described in [Issue: Improve UX of Step Resource Requests](https://github.com/tektoncd/pipeline/issues/2986).
+Specifically, all `Step` resource requests other than the maximum are ignored.
+
+2. We don't support parameterizing Step and Sidecar resource requirements. Even though we don't claim to support this,
+some users perceive the ability to parameterize some Task fields but not others [as a bug](https://github.com/tektoncd/community/pull/560#issuecomment-993348697).
+This feature was requested in [FR: Allow usage of variable replacement when defining resource limits and requests](https://github.com/tektoncd/pipeline/issues/4080).
+
+3. We don't support specifying resource requirements at runtime. Parameterizing resource requirements would not fully address this problem,
+as users should be able to specify resource requirements for Tasks they don't own, such as Catalog Tasks.
+This issue is described in more detail in [FR: Support specifying resource requests at TaskRun level](https://github.com/tektoncd/pipeline/issues/4326)
+
+This section elaborates on these issues.
+
+### Confusing Task Resource Requirements UX
+Under the hood, Tekton modifies the resource requests users specify to account for how the Tekton entrypoint runs each `Step` container.
+These modifications aren't transparent to users and have caused confusion.
+
+#### Kubernetes pod scheduling
+Kubernetes runs each init container in sequence before a pod starts up, and then runs containers and sidecars in parallel.
+A pod's [effective resource requests and limits](https://kubernetes.io/docs/concepts/workloads/pods/init-containers/#resources) are the higher of:
+
+- the sum of container and sidecar resource requests/limits
+- the maximum requests/limits of any init container
+
+This information is used to schedule the pod on a node, and the pod may not be scheduled if its requirements are too large.
+Cluster admins may also define [resource quotas](https://kubernetes.io/docs/concepts/policy/resource-quotas/), which restrict resource consumption
+per namespace, and [limit ranges](https://kubernetes.io/docs/concepts/policy/limit-range/), which restrict resource consumption per pod, container, or PVC.
+
+#### Tekton Task scheduling
+Tekton schedules each `Task` on its own pod and each `Step` in its own container. While Kubernetes runs each container in parallel,
+Tekton `Step`s are run sequentially. This means that the amount of resources needed by the `Task`'s pod is constrained by the maximum `Step`
+resource requests, rather than the sum of the resource requests of all `Step`s.
+
+Tekton implements this by applying the maximum `Step` resource request to one container, and not applying resource requests to any other container.
```

Comment:
This is not true anymore though, it doesn't zero out resource request anymore, it uses what the user gave *or* use LimitRange but even there, never zeros out anything.